rust/main/hyperlane-base/src/contract_sync/mod.rs

```diff
@@ -397,9 +397,17 @@ where
         &self,
         index_settings: IndexSettings,
     ) -> Result<Box<dyn ContractSyncCursor<T>>> {
-        let watermark = self.store.retrieve_high_watermark().await.unwrap();
+        let watermark: Option<u32> = self.store.retrieve_high_watermark().await.unwrap();
```

Comment:
any reason for us to unwrap here instead of `?` ?

Reply:
It is a good question. I actually don't want to change this line. `unwrap()` is there already.

I'll check if we can just return with `?`.

Reply:
We are handling error at the place where we call this method. So, exiting with `?` is okay. I'll fix it.